frontend/app/components/input-sin-field.tsx

```diff
@@ -0,0 +1,79 @@
+import { PatternFormat } from 'react-number-format';
+
+import { InputError } from './input-error';
+import { InputHelp } from './input-help';
+import { InputLabel } from '~/components/input-label';
+import { cn } from '~/utils/tw-utils';
+
+const inputBaseClassName = 'block rounded-lg border-gray-500 focus:border-blue-500 focus:outline-none focus:ring focus:ring-blue-500';
+const inputDisabledClassName = 'disabled:bg-gray-100 disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70';
+const inputReadOnlyClassName = 'read-only:bg-gray-100 read-only:pointer-events-none read-only:cursor-not-allowed read-only:opacity-70';
+const inputErrorClassName = 'border-red-500 focus:border-red-500 focus:ring-red-500';
+
+export interface InputSinFieldProps extends Omit<React.ComponentProps<typeof PatternFormat>, 'aria-errormessage' | 'aria-invalid' | 'aria-labelledby' | 'aria-required' | 'children' | 'value' | 'onChange' | 'format'> {
+  defaultValue: string;
+  errorMessage?: string;
+  helpMessagePrimary?: React.ReactNode;
+  helpMessagePrimaryClassName?: string;
+  helpMessageSecondary?: React.ReactNode;
+  helpMessageSecondaryClassName?: string;
+  id: string;
+  label: string;
+  locale?: string;
```

Comment:
Is `locale` needed? I don't see it being used anywhere.

Reply:
Fixed